src/commands/manifest.ts

```diff
@@ -99,16 +113,30 @@ export default class Manifest extends Command {
     this.log(`wrote manifest to ${file}`)
   }
 
-  private cloneRepo(repoUrl: string, fullPath: string, tag: string | semver.SemVer | null): void {
-    try {
-      this.executeCommand(`git clone --branch ${tag} ${repoUrl} ${fullPath} --depth 1`)
-    } catch {
-      try {
-        this.executeCommand(`git clone --branch v${tag} ${repoUrl} ${fullPath} --depth 1`)
-      } catch {
-        throw new Error(`Unable to clone repo ${repoUrl} with tag ${tag}`)
+  private getTarballUrl(plugin: string, version: string): string {
+    // jit plugin is unpinned so we need to figure out the max satisfying version
+    if ((version.startsWith('^') || version.startsWith('~'))) {
+      const npmLatest = JSON.parse(this.executeCommand(`npm view ${plugin}@latest --json`).stdout) as {
+        versions: string[]
+        dist: { tarball: string }
       }
+      const maxSatisfying = semver.maxSatisfying(npmLatest.versions, version)
+
+      const {dist} = JSON.parse(this.executeCommand(`npm view ${plugin}@${maxSatisfying} --json`).stdout) as {
+        versions: string[]
+        dist: { tarball: string }
+      }
+
+      return dist.tarball
     }
+
+    // jit plugin is pinned so we don't need to figure out the max satisfying version
+    const {dist} = JSON.parse(this.executeCommand(`npm view ${plugin}@${version} --json`).stdout) as {
+      versions: string[]
+      dist: { tarball: string }
+    }
+    return dist.tarball
+
```

Comment:
this is a bit repetitive.

suggestion: 
- 1 function to calculate the version (pinned version if pinned, use your new `maxSatisfying` technique otherwise)
- 1 function to get the tarball, given a version